src/Umbraco.Abstractions/Persistence/Repositories/IKeyValueRepository.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Umbraco.Infrastructure.Persistence.Repositories
+{
+    public interface IKeyValueRepository
```

Comment:
This doesn't really follow any of the current repository patterns that we have. All repsositories implement at least `IRepository`, then we have `IReadRepository`, `IWriteRepository`, and `IQueryRepository` which are all implemented by `IReadWriteQueryRepository`. Whether it makes sense here to implement all of them I'm unsure but we should try to be consistent. Most repositories inherit from `RepositoryBase` which has the normal plumbing wired up for Scopes, etc... but requires your entity to be of type `IEntity`. You could even return a consistent GUID Key for the entity based on the `string` key using the string extension `ToGuid` method that we have.  We should also have a task to update this DB table to have both a Created and Updated date time so we can be more consistent with the EntityBase model. Perhaps go as far as to add a real GUID column to the DB table so it's actually get/set instead of just get.

Reply:
Yes, point noted.  I've updated the `IKeyValueRepository` to use these base interfaces, and make the amends that follow from that.  It's slightly clunky in the sense that the "identifier" for this entity is the key of the key value, but `Key` is already defined on the `IEntity`.  But follows existing patterns better as you noted.